pinot-core/src/main/java/org/apache/pinot/core/transport/server/routing/stats/ServerRoutingStatsManager.java

```diff
@@ -377,4 +389,15 @@ public Double fetchHybridScoreForServer(String server) {
       stats.getServerReadLock().unlock();
     }
   }
+
+  private void recordQueueSizeMetrics() {
+    int queueSize = getQueueSize();
+    _brokerMetrics.setValueOfGlobalGauge(BrokerGauge.ROUTING_STATS_MANAGER_QUEUE_SIZE, queueSize);
+    if (queueSize > _executorQueueSizeWarnThreshold) {
+      _brokerMetrics.addMeteredGlobalValue(BrokerMeter.ROUTING_STATS_MANAGER_Q_LIMIT_REACHED, 1L);
```

Comment:
I think emitting the warning metric is not needed. We already have a gauge. Users should set warning/critical alerts on their end. Though the gauge value will get overwritten frequently and we will only have 1 value per bucket, I think it should be good enough since this is only measuring the stats.

Reply:
Thanks Ankit, addressed